tests/test_sensors/test_led.py

```diff
@@ -0,0 +1,69 @@
+import pytest
+from src.sensors.led import FixedLED
+
+@pytest.fixture
+def init_led():
+    return FixedLED(None, 0, 0)
+
+@pytest.mark.parametrize("led, red, green, blue, expected", [
+    ("init_led", 0, 0, 0, 0),
+    ("init_led", 1, 0, 0, 1),
+    ("init_led", 5, 2, 0, 5),
+    ("init_led", 2, 6, 0, 2)
+])
+def test_set_colour_red(led, red, green, blue, expected, request):
+    led = request.getfixturevalue(led)
+    led.set_colour(red, green, blue)
+    assert led.red_value == expected
+
+@pytest.mark.parametrize("led, red, green, blue, expected", [
+    ("init_led", 0, 0, 0, 0),
+    ("init_led", 0, 1, 0, 1),
+    ("init_led", 2, 5, 0, 5),
+    ("init_led", 0, 2, 6, 2)
+])
+def test_set_colour_green(led, red, green, blue, expected, request):
+    led = request.getfixturevalue(led)
+    led.set_colour(red, green, blue)
+    assert led.green_value == expected
+
+@pytest.mark.parametrize("led, red, green, blue, expected", [
+    ("init_led", 0, 0, 0, 0),
+    ("init_led", 0, 0, 1, 1),
+    ("init_led", 0, 2, 5, 5),
+    ("init_led", 5, 0, 2, 2)
+])
+def test_set_colour_blue(led, red, green, blue, expected, request):
+    led = request.getfixturevalue(led)
+    led.set_colour(red, green, blue)
+    assert led.blue_value == expected
+    
+def test_turn_off(init_led):
+    init_led.turn_off
+    assert (init_led.red_value == 0 and init_led.blue_value == 0 and init_led.green_value == 0)
+
+def test_get_color(init_led):
+    init_led.set_colour(0, 0, 0)
+    assert init_led.get_colour() == (0, 0, 0)
+
+@pytest.mark.parametrize("led, xval", [
+    ("init_led", 0),
+    ("init_led", 2),
+    ("init_led", -1),
+])
+def test_set_xvalue(led, xval, request):
+    led = request.getfixturevalue(led)
+    led.set_xvalue(xval)
+    assert led.x == xval
+
+@pytest.mark.parametrize("led, yval", [
+    ("init_led", 0),
+    ("init_led", 2),
+    ("init_led", -1),
+])
+def test_set_yvalue(led, yval, request):
```

Comment:
Could these two last tests be combined into one? It seems to be a waste that we are just running slightly different tests with the same parameters.

Reply:
I'd argue otherwise. I know this is basic stuff, it's more for my learning of pytest, but wouldn't we want to isolate each function? The second to last tests the `set_xvalue()` function while the last does `set_yvalue()`. Is the point of unit tests not to isolate functions?